### PR TITLE
fix(@desktop/communities): improve member list sorting

### LINKED ISSUE #3163: improve member list sorting
The user list currently can be a bit confusing as it's sorted somewhat randomly. We should improve it so:

- online members show the green circle
- offline members don't show any circle
- user should be grouped with online users and offline users https://www.figma.com/file/Zz6KMBxwrB0l2NPgsjnPjq/%F0%9F%91%A4-Profile%E2%8E%9CDesktop?node-id=2816%3A15806
- list should be ordered 2) users with ens names 3) alphabetical

Screen Shot 2021-08-04 at 6 06 28 PM

### PR DESCRIPTION
**Pull request by pinebit**
Fixes #3163

### CODE REVIEW

File: ui/main.qml [hunk 1]
```diff
@@ -42,8 +42,8 @@ StatusWindow {
     objectName: "mainWindow"
     minimumWidth: 900
     minimumHeight: 600
-    width: 1232
-    height: 770
+    width: Math.min(1232, Screen.desktopAvailableWidth - 64)
```
Comment (pinebit) [inline ui/main.qml:45]: this is not strictly related to the issue, but it was so annoying on my laptop (4K @ 250% scale) - the window spans beyond my desktop area.

File: ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml [hunk 2]
```diff
@@ -27,6 +27,31 @@ Item {
...
+        for (let r = 0; r < community.members.rowCount(); r++) {
+            const pubKey = community.members.rowData(r, "address")
+            const nickname = appMain.getUserNickname(pubKey)
+            const identicon = community.members.rowData(r, "identicon")
+            const ensName = community.members.rowData(r, "ensName")
+            const name = !ensName.endsWith(".eth") && !!nickname ? nickname : Utils.removeStatusEns(ensName)
+            const statusType = chatsModel.communities.activeCommunity.memberStatus(pubKey)
+            const lastSeen = chatsModel.communities.activeCommunity.memberLastSeen(pubKey)
+            const lastSeenMinutesAgo = (currentTime / 1000 - parseInt(lastSeen)) / 60
+            const online = (pubKey === profileModel.profile.pubKey) || (lastSeenMinutesAgo < 7)
```
Comment (alaibe) [inline ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml:41]: Is it correct to assume we are always online? even if we want to be not visible?

File: ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml [hunk 3]
```diff
@@ -27,6 +27,31 @@ Item {
...
+
+            proxyModel.append({
+                pubKey: pubKey,
+                name: name,
+                identicon: identicon,
+                lastSeen: lastSeen,
+                statusType: statusType,
+                online: online,
+                sortKey: "%1%2".arg(online ? "A" : "B").arg(name)
```
Comment (alaibe) [inline ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml:50]: Why not doing this nim side, that would avoid us to go through the list and create a new one?
Wouldn't that be more performant?

**Review by iurimatias — CHANGES_REQUESTED** [review]
Screen Shot 2021-08-18 at 10 20 51 AM

online tag is repeated

File: ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml (hunk 2 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml:41]: This logic is borrowed from User.qml, and I assumed if it was done this way, then this is expected behavior.

**Comment by pinebit** [comment]
Rebased and fixed

File: ui/app/AppLayouts/Chat/ChatColumn/User.qml [hunk 4]
```diff
@@ -11,6 +11,7 @@ Item {
     property string lastSeen: ""
     property string identicon
     property int statusType: Constants.statusType_Online
+    property color offlineColor: Style.current.darkGrey
```
Comment (alaibe) [inline ui/app/AppLayouts/Chat/ChatColumn/User.qml:14]: Is it working in dark mode, I had to change this to midGrey in other places

**Review by alaibe — APPROVED** [review]

File: ui/app/AppLayouts/Chat/ChatColumn/User.qml (hunk 4 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/ChatColumn/User.qml:14]: I did not make it darkGrey, I only made it conditionally transparent.

File: ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml (hunk 3 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml:50]: Will try re-implementing it that way today.

File: ui/app/AppLayouts/Chat/ChatColumn/User.qml (hunk 4 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/ChatColumn/User.qml:14]: I will check it though in dark mode.

Comment (alaibe) [reply ui/app/AppLayouts/Chat/ChatColumn/User.qml:14]: Yeah, I know it was like that before, just saw you modified and fix something similar yesterday, so only pinging ;)
Feel free to ignore

File: ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml (hunk 3 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/CommunityComponents/CommunityUserList.qml]: @alaibe all done, please verify!

**Review by alaibe — APPROVED** [review]
LGTM, nice one
